### PR TITLE
fix: Include management-fee delta in doOverpayment assertion

### PR DESCRIPTION
**Pull request by Tapanito**
Two related issues on the loan overpayment path triggered assertion failures.

1. computeOverpaymentComponents requires its input amount to be  representable at the loan's scale (`isRounded(asset, overpayment,  loanScale)`). The residual passed in by loanMakePayment was the raw `amount - totalPaid` (capped at totalValueOutstanding), which could carry more precision than loanScale allows when the borrower pays an amount with extra fractional digits. Round the residual down to loanScale before calling computeOverpaymentComponents.

2. The "interest paid agrees" assertion in doOverpayment was missing the management-fee component released during re-amortization. When a borrower overpays and the loan is re-amortized, the management fee outstanding decreases, but the original assertion formula did not account for this delta. Add `mfeeReleased` (oldMfee - newMfee) to the assertion, and introduce named intermediate variables with  a comment explaining the loan-state identity behind the derivation.

Add a regression test (testBugOverpayUnroundedAmount) that exercises the overpayment path with a non-zero management fee rate and an unrounded payment amount.

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

**Comment by godexsoft** [comment]
We recently merged a refactor to `develop` that enables clang-tidy's `readability-identifier-naming`. Your branch now has heavy conflicts that are largely mechanical. Below is a workflow that aligns your branch's naming with develop _before_ merging, which should minimize the merge conflicts.

## One-time setup

If you don't already have clang-tidy working in your env, on macOS:

```bash
brew install llvm@21
# Follow brew's hint to put $(brew --prefix llvm@21)/bin on PATH so run-clang-tidy is found.
```

## Workflow on your branch (before merging develop)

**1. Grab the new `.clang-tidy` from develop without pulling anything else.** Sync your fork on GitHub first, then:

```bash
git remote -v   # should show 'upstream' among others; if not:
# git remote set-url upstream git@github.com:XRPLF/rippled.git
git fetch upstream
git checkout upstream/develop -- .clang-tidy
```

**2. Reconfigure conan/cmake so `compile_commands.json` is fresh.**

**3. Apply renames for the files modified in your PR:**

```bash
git diff --name-only $(git merge-base HEAD upstream/develop) HEAD \
  | grep -E '\.(cpp|h|hpp|ipp)$' \
  | xargs run-clang-tidy -p build -fix -allow-no-checks
# or -p .build, or whatever your build dir is called
```

**4. Build + test, then commit as a single dedicated commit:**

```bash
cmake --build build -j8
git commit -am "refactor: Align identifier naming with develop"
```

**5. Now merge develop:**

```bash
git merge upstream/develop
```

## Extra

Run clang-tidy once more after the merge to catch any stragglers introduced from develop's side:

```bash
run-clang-tidy -p build -fix -allow-no-checks src tests
# or -p .build, or whatever your build dir is called
```

File: src/libxrpl/ledger/helpers/LendingHelpers.cpp [hunk 1]
```diff
@@ -1878,50 +1892,57 @@ loanMakePayment(
         // It shouldn't be possible for the overpayment to be greater than
         // totalValueOutstanding, because that would have been processed as
         // another normal payment. But cap it just in case.
-        Number const overpayment = std::min(amount - totalPaid, *totalValueOutstandingProxy);
+        Number const overpaymentRaw = std::min(amount - totalPaid, *totalValueOutstandingProxy);
+        Number const overpayment = view.rules().enabled(fixCleanup3_2_0)
+            ? roundToAsset(asset, overpaymentRaw, loanScale, Number::RoundingMode::Downward)
+            : overpaymentRaw;
```
Comment (a1q123456) [inline src/libxrpl/ledger/helpers/LendingHelpers.cpp:2050]: Better to also test the case where fixCleanup3_2_0 is not enabled. Currently, codecov says the line 1898 isn't tested.

File: src/libxrpl/ledger/helpers/LendingHelpers.cpp [hunk 2]
```diff
@@ -1878,50 +1892,57 @@ loanMakePayment(
...
-                overpayment,
-                overpaymentInterestRate,
-                overpaymentFeeRate,
-                managementFeeRate);
-
-        // Don't process an overpayment if the whole amount (or more!)
-        // gets eaten by fees and interest.
-        if (overpaymentComponents.trackedPrincipalDelta > 0)
+        // Due to rounding, overpayment could be zero
+        if (overpayment > 0)
```
Comment (a1q123456) [inline src/libxrpl/ledger/helpers/LendingHelpers.cpp]: I think we should bypass this check when the fix isn't enabled.

**Review by a1q123456 — COMMENTED** [review]
Left some suggestions

File: src/libxrpl/ledger/helpers/LendingHelpers.cpp (hunk 2 above)
Comment (Tapanito) [reply src/libxrpl/ledger/helpers/LendingHelpers.cpp]: Done.

File: src/libxrpl/ledger/helpers/LendingHelpers.cpp (hunk 1 above)
Comment (Tapanito) [reply src/libxrpl/ledger/helpers/LendingHelpers.cpp:2050]: Done.

**Review by godexsoft — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -86,3 +86,5 @@ __pycache__
 
 # clangd cache
 /.cache
+.codedb
+.claude/settings.json
```
Comment (bthomee) [inline .gitignore]: Can this be moved to the "AI tools" section? Also,`.claude` is already excluded, so adding a subpath does not change anything and this entry can be removed.

Comment (Tapanito) [reply .gitignore]: Thanks!! This was meant to be in `.gitignore_local` in the first-place. Removed.

File: src/test/app/Loan_test.cpp [hunk 4]
```diff
@@ -169,6 +169,10 @@ class Loan_test : public beast::unit_test::Suite
         TenthBips32 coverRateLiquidation = percentageToTenthBips(25);
         std::string data = {};  // NOLINT(readability-redundant-member-init)
         std::uint32_t flags = 0;
+        // If set, the vault is created with this sfScale value. Useful for
+        // tests that need finer loanScale to exercise rounding edge cases.
+        std::optional<std::uint32_t> vaultScale =
```
[the remark this answers is not in the document]
Comment (a1q123456) [reply src/test/app/Loan_test.cpp]: I think this makes sense.

[the remark this answers is not in the document]
Comment (Tapanito) [reply src/test/app/Loan_test.cpp]: fixed.

**Review by a1q123456 — APPROVED** [review]
LGTM

File: src/libxrpl/ledger/helpers/LendingHelpers.cpp [hunk 5]
```diff
@@ -783,11 +780,28 @@ doOverpayment(
...
+    //        newTVO - (oldTVO - dp)
+    //   2. Management fee released by re-amortization (positive when
+    //      mfee decreased; zero when managementFeeRate == 0):
+    //        oldMfee - newMfee
+    //   3. The overpayment's penalty interest part (= untrackedInterest
+    //      for the overpayment path; see computeOverpaymentComponents):
+    //        trackedInterestPart()
+    [[maybe_unused]] Number const tvoChange = newRoundedLoanState.valueOutstanding -
+        (totalValueOutstandingProxy - overpaymentComponents.trackedPrincipalDelta);
+    [[maybe_unused]] Number const mfeeReleased =
```
Comment (bthomee) [inline src/libxrpl/ledger/helpers/LendingHelpers.cpp]: Let's just rename `mFeeReleased` to `managementFeeReleased`. It's consistent with the other variable names, and doesn't cause potential confusion with those who think it refers to a class member variable.

Comment (Tapanito) [reply src/libxrpl/ledger/helpers/LendingHelpers.cpp]: Done.
